adafruit_emc2101/__init__.py

```diff
@@ -340,12 +376,20 @@ def tach_limit(self):
 
         low = self._tach_limit_lsb
         high = self._tach_limit_msb
-
-        return _FAN_RPM_DIVISOR / ((high << 8) | low)
+        limit = high << 8 | low
+        if limit < 2:
+            print("read tach limit 0")
+            return 0
+        return emc2101_regs._FAN_RPM_DIVISOR / limit
 
     @tach_limit.setter
     def tach_limit(self, new_limit):
-        num = int(_FAN_RPM_DIVISOR / new_limit)
+        """Set the speed limiter on the fan PWM signal. The value of 15000 is
+        arbitrary, but very few fans run faster than this.
+        """
+        if new_limit not in range(1, 15000):
```

Comment:
It looks like you probably meant `range(1, 14001)`.  I also can remember if using `range` this way will eat up a lot of memory.  I'll look into this, but it may be better to do the following to explicitly check the bounds only:

```suggestion
        if not 1 <= new_limit <= 14000:
```

Reply:
Okay, I checked, it's better to use the explicit check instead of `range`!

Reply:
Unresolving because the explicit check (the one in the suggestion) should be used.